src/input/InputEventMapper.cc

```diff
@@ -149,16 +149,11 @@ int InputEventMapper::parseSettingValue(const std::string val)
 void InputEventMapper::onInputMappingUpdated()
 {
     Settings::Settings *s = Settings::Settings::inst();
-    actions[0] = QString(s->get(Settings::Settings::inputButton0).toString().c_str());
-    actions[1] = QString(s->get(Settings::Settings::inputButton1).toString().c_str());
-    actions[2] = QString(s->get(Settings::Settings::inputButton2).toString().c_str());
-    actions[3] = QString(s->get(Settings::Settings::inputButton3).toString().c_str());
-    actions[4] = QString(s->get(Settings::Settings::inputButton4).toString().c_str());
-    actions[5] = QString(s->get(Settings::Settings::inputButton5).toString().c_str());
-    actions[6] = QString(s->get(Settings::Settings::inputButton6).toString().c_str());
-    actions[7] = QString(s->get(Settings::Settings::inputButton7).toString().c_str());
-    actions[8] = QString(s->get(Settings::Settings::inputButton8).toString().c_str());
-    actions[9] = QString(s->get(Settings::Settings::inputButton9).toString().c_str());
+    for (int i = 0; i < 10; i++ ){
+		std::string is = std::to_string(i);
+		Settings::SettingsEntry* ent =s->getSettingEntryByName("button" +is);
+		actions[i] =(s->get(*ent).toString().c_str());
```

Comment:
This may be a bit nicer: `actions[i] = QString::fromStdString(s->get(*ent).toString())`

Reply:
Do you really want to use a QT-Functions where we are not interacting with the UI/QT?
Sure, using the input-driver without UI would not make sense, but still: The InputEventMapper is low-level code that could be used independent of GUI framework beeing used.

Reply:
In general not, but in this case, `actions` is an array of `QString`:
```
    QString actions[10];
```

Reply:
understood, thank you.